features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/CreateScope.jsx

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import Grid from '@material-ui/core/Grid';
+import TextField from '@material-ui/core/TextField';
+import React from 'react';
+import APIPropertyField from 'AppComponents/Apis/Details/Overview/APIPropertyField';
+import Typography from '@material-ui/core/Typography';
+import { FormattedMessage } from 'react-intl';
+import TagsInput from 'react-tagsinput';
+import Api from 'AppData/api';
+import Button from '@material-ui/core/Button';
+import PropTypes from 'prop-types';
+import { message } from 'antd/lib/index';
+import { withRouter } from 'react-router';
+import { Link } from 'react-router-dom';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = theme => ({
+    buttonSave: {
+        marginTop: theme.spacing.unit * 10,
+    },
+    buttonCancel: {
+        marginTop: theme.spacing.unit * 10,
+        marginLeft: theme.spacing.unit * 5,
+    },
+    topics: {
+        marginTop: theme.spacing.unit * 10,
+    },
+    headline: {
+        paddingTop: theme.spacing.unit * 1.5,
+        paddingLeft: theme.spacing.unit * 2.5,
+    },
+});
+
+/**
+ * Create new scoped for an API
+ * @class CreateScope
+ * @extends {Component}
+ */
+
+class CreateScope extends React.Component {
+    constructor(props) {
+        super(props);
+        this.api = new Api();
+        this.api_uuid = props.match.params.api_uuid;
+        this.state = {
+            apiScopes: null,
+            apiScope: {},
+            roles: [],
+        };
+        this.addScope = this.addScope.bind(this);
+        this.handleInputs = this.handleInputs.bind(this);
+    }
+
+    /**
+     * Add new scope
+     * @memberof Scopes
+     */
+    addScope() {
+        const hideMessage = message.loading('Adding the Scope ...', 0);
```

Comment:
@kavishkafernando do we support for internationalization for these phrases? If not let's create a high priority github issue as we need to address it as soon as possible